data-prepper-plugins/mutate-event-processors/src/main/java/org/opensearch/dataprepper/plugins/processor/mutateevent/AddEntryProcessorConfig.java

```diff
@@ -14,10 +14,42 @@
 import jakarta.validation.constraints.NotEmpty;
 import jakarta.validation.constraints.NotNull;
 import org.opensearch.dataprepper.model.annotations.AlsoRequired;
+import org.opensearch.dataprepper.model.annotations.ConditionalRequired;
 
 import java.util.List;
 import java.util.stream.Stream;
 
+@ConditionalRequired(value = {
+        @ConditionalRequired.IfThenElse(
+                ifFulfilled = {@ConditionalRequired.SchemaProperty(field = "key", value = "null")},
+                thenExpect = {@ConditionalRequired.SchemaProperty(field = "metadata_key")}
+        ),
+        @ConditionalRequired.IfThenElse(
+                ifFulfilled = {@ConditionalRequired.SchemaProperty(field = "metadata_key", value = "null")},
+                thenExpect = {@ConditionalRequired.SchemaProperty(field = "key")}
+        ),
+        @ConditionalRequired.IfThenElse(
+                ifFulfilled = {
+                        @ConditionalRequired.SchemaProperty(field = "format", value = "null"),
+                        @ConditionalRequired.SchemaProperty(field = "value", value = "null"),
+                },
+                thenExpect = {@ConditionalRequired.SchemaProperty(field = "value_expression")}
+        ),
+        @ConditionalRequired.IfThenElse(
+                ifFulfilled = {
+                        @ConditionalRequired.SchemaProperty(field = "format", value = "null"),
+                        @ConditionalRequired.SchemaProperty(field = "value_expression", value = "null"),
```

Comment:
Is there a way to express `OR` releation when there are more than one condition grouped under `ifFulfilled`

Reply:
For OR relation we have to use separate IfThenElse annotation

Reply:
goit 👍